src/mono/System.Private.CoreLib/src/System/RuntimeType.Mono.cs

```diff
@@ -1127,8 +1118,124 @@ public override MemberInfo[] GetMember(string name, MemberTypes type, BindingFla
 
             return compressMembers;
         }
-        #endregion
 
+        public override MemberInfo GetMemberWithSameMetadataDefinitionAs(MemberInfo member)
+        {
+            if (member is null) throw new ArgumentNullException(nameof(member));
+
+            MemberInfo? result = member.MemberType switch
+            {
+                MemberTypes.Method => GetMethodWithSameMetadataDefinitionAs(member),
+                MemberTypes.Constructor => GetConstructorWithSameMetadataDefinitionAs(member),
+                MemberTypes.Property => GetPropertyWithSameMetadataDefinitionAs(member),
+                MemberTypes.Field => GetFieldWithSameMetadataDefinitionAs(member),
+                MemberTypes.Event => GetEventWithSameMetadataDefinitionAs(member),
+                MemberTypes.NestedType => GetNestedTypeWithSameMetadataDefinitionAs(member),
+                _ => null
+            };
+
+            return result ?? throw CreateGetMemberWithSameMetadataDefinitionAsNotFoundException(member);
+        }
+
+        private const BindingFlags GetMemberWithSameMetadataDefinitionAsBindingFlags =
+            BindingFlags.Instance | BindingFlags.Static | BindingFlags.Public | BindingFlags.NonPublic;
+
+        private MemberInfo? GetMethodWithSameMetadataDefinitionAs(MemberInfo methodInfo)
+        {
```

Comment:
Does this also needs the loop that walks up the inheritance hiearchy?

Reply:
Fixed.